content/docs/reference/service-urls.md

```diff
@@ -0,0 +1,287 @@
+  - authenticate internal service url
+  - authenticate_service_url
+  - authenticate_internal_service_url
+  - authorize service url
+  - authorize_service_url
+  - authorize internal service url
+  - authorize_internal_service_url
+pagination_prev: null
+pagination_next: null
+toc_max_heading_level: 2
+---
+
+import Tabs from '@theme/Tabs'; import TabItem from '@theme/TabItem'; import PomeriumAuthenticateServices from '@site/content/docs/admonitions/\_pomerium-authenticate-services.mdx';
+
+# Service URL Settings
+
+## Authenticate Service URL
+
+The **Authenticate Service URL** setting defines the externally accessible URL where Pomerium redirects end users (clients) to authenticate against an identity provider.
+
+If Pomerium is running in [split-service mode](/docs/internals/configuration#all-in-one-vs-split-service-mode), each Pomerium service requires the authenticate service URL in its configuration.
+
+:::info
+
+<PomeriumAuthenticateServices />
+
+:::
+
+### How to configure
+
+<Tabs>
+<TabItem value="Core" label="Core">
+
+| **Config file keys** | **Environment variables** | **Type** | **Usage** |
+| :-- | :-- | :-- | :-- |
+| `authenticate_service_url` | `AUTHENTICATE_SERVICE_URL` | `URL` | \***required** |
+
+\* Excluding the `authenticate_service_url` defaults to the [hosted authenticated service](/docs/capabilities/hosted-authenticate-service).
+
+### Examples
+
+```yaml
+authenticate_service_url: https://authenticate.corp.example.com
+```
+
+```bash
+AUTHENTICATE_SERVICE_URL=https://authenticate.corp.example.com
+```
+
+</TabItem>
+<TabItem value="Enterprise" label="Enterprise">
+
+`authenticate_service_url` is a bootstrap configuration setting and is not configurable in the Console.
+
+</TabItem>
+<TabItem value="Kubernetes" label="Kubernetes">
+
+| **[Parameter name](/docs/k8s/reference#spec** | **Type** | **Usage**      |
+| :-------------------------------------------- | :------- | :------------- |
+| `authenticate.url`                            | `URL`    | \***required** |
+
+\* Excluding the `authenticate.url` defaults to the [hosted authenticated service](/docs/capabilities/hosted-authenticate-service).
+
+### Examples
+
+```yaml
+authenticate:
+  url: https://authenticate.corp.example.com
+```
+
+See the [Kubernetes - Global Configuration](/docs/k8s/configure) for more information.
+
+</TabItem>
+</Tabs>
+
+## Authenticate Internal Service URL
+
+The **Authenticate Internal Service URL** setting defines an internally accessible URL where Pomerium redirects end users (clients) to authenticate against an identity provider. Use this setting if Pomerium can't access an external service URL.
+
+You may include both an internal and external authenticate service URL in the same configuration. In this case, the internal authenticate service URL overrides the external service URL when Pomerium determines which TLS certificate and hostname the authenticate service will listen with when receiving incoming client requests.
```

Comment:
I don't think the part about "overrides the external service URL when Pomerium determines which TLS certificate and hostname the authenticate service will listen" is accurate.

The internal authenticate service URL is used when another Pomerium service wants to communicate with the authenticate service.

Reply:
So, I got that from the previous definition from v0.21, the version before these reference pages were updated. (See [Authenticate Internal Service URL](https://0-21-0.docs.pomerium.com/docs/reference/authenticate-internal-service-url).) 

Can we get clarity on what actually would happen if you included both URLs?

Reply:
This is my understanding: if you configure both Authenticate Service URL and Authenticate Internal Service URL, then the authenticate service will use the Authenticate Service URL for all the login / logout redirects, but other Pomerium services will use the Authenticate Internal Service URL when attempting to communicate with the authenticate service. (Although this may be less relevant since we restored the older authentication flow when not using the hosted authenticate service.)

It would probably be a good exercise to set up an environment with another proxy in front of Pomerium such that you would need to set the internal service URLs.